cocos/asset/assets/simple-texture.ts

```diff
@@ -140,7 +142,7 @@ export class SimpleTexture extends TextureBase {
      * @param level @en Mipmap level to upload the image to. @zh 要上传的 mipmap 层级。
      * @param arrayIndex @en The array index. @zh 要上传的数组索引。
      */
-    public uploadData (source: HTMLCanvasElement | HTMLImageElement | ArrayBufferView | ImageBitmap, level = 0, arrayIndex = 0): void {
+    public uploadData (source: ImageSource | ArrayBufferView | ImageAsset, level = 0, arrayIndex = 0): void {
```

Comment:
Could we deprecate `uploadData` here?
Add a new function which access `(data: ImageData, level = 0, arrayIndex = 0)`

Reply:
Yes, I've overloaded it.